R/module_nested_tabs.R

```diff
@@ -255,13 +256,32 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
     }
   )
 
+  hashes <- calculate_hashes(datanames, datasets)
```

Comment:
I think the hashes should be calculated earlier than this as here we are recalculating hashes for each module separately as they will all have different datanames - I think `.datasets_to_data` should have an extra argument hashes which is a named list of hashes for all datasets and then get_datasets_code only outputs the hashes of those needed anyway?

Reply:
though on the other hand if hashing is quick then the extra complexity of moving it up the chain is probably not worth it...

Reply:
The good place to calculate the hashes once would be in `modules_tabs_with_filters` but we will then have to calculate the hashes twice in `ui_modules_tabs_with_filters` and `srv_modules_tabs_with_filters` and propagate them as arguments in both `ui_srv_nested_tabs` and `srv_nested_tabs` as `.datasets_to_data` is called in both `ui_srv_nested_tabs.teal_module` and `srv_nested_tabs.teal_module`.
We decided with Nik that we will keep the implementation as it is now as the calculation cost is negligible.